closure/compiler/closure_js_binary.bzl

```diff
@@ -101,6 +101,8 @@ def _impl(ctx):
              "--define=goog.json.USE_NATIVE_JSON"]
   if ctx.attr.output_wrapper:
     args += ["--output_wrapper=%s" % ctx.attr.output_wrapper]
+    if ctx.attr.output_wrapper == "(function(){%output%}).call(this);":
```

Comment:
Another example of an output wrapper would be `"%output%\n//# sourceMappingURL=foo_bin.js.map"`. But that's obviously not inside a function.

I would prefer not to special case this unless we have to. Would it be possible to solve this algorithmically? It almost reminds me of a popular interview question.

For posterity:

```
 --assume_function_wrapper              : Enable additional optimizations based
                                          on the assumption that the output
                                          will be wrapped with a function
                                          wrapper.  This flag is used to
                                          indicate that "global" declarations
                                          will not actually be global but
                                          instead isolated to the compilation
                                          unit. This enables additional
                                          optimizations.
```